### PR TITLE
Revises memory manager

### PR DESCRIPTION
**Pull request by ultimaweapon**
With this version we don't need a custom allocator anymore because we can map `mmap` and `munmap` to the host API directly.

### CODE REVIEW

**Comment by bayedieng** [comment]
It's probably best to have system calls in Linux and macOS targeted differently since macOS tends to have some slight variations to POSIX.

**Comment by Mou-Ikkai** [comment]
Also, I would say to keep the ReadME memory requirement. The PS4 has 8GB of RAM, which we can't just make that magically smaller.

File: src/kernel/src/memory/mod.rs [hunk 1]
```diff
@@ -1,61 +1,53 @@
...
+        // granularity because it is always multiply of page size, which is a correct value.
+        let (page_size, allocation_granularity) = Self::get_memory_model();
 
         if page_size > Self::VIRTUAL_PAGE_SIZE || (Self::VIRTUAL_PAGE_SIZE % page_size) != 0 {
+            // If page size is larger than PS4 we will have a problem with memory protection.
+            // Let's say page size on the host is 32K and we have 2 adjacent virtual pages, which is
+            // 16K per virtual page. The first virtual page want to use read/write while the second
+            // virtual page want to use read-only. This scenario will not be possible because those
+            // two virtual pages are on the same page.
             panic!("Your system are using unsupported page size.");
```
Comment (Mou-Ikkai) [inline src/kernel/src/memory/mod.rs:31]:
```suggestion
            panic!("Your system is using an unsupported page size.");
```

**Review by Mou-Ikkai — COMMENTED** [review]
Small grammar change.

**Comment by ultimaweapon** [comment]
@bayedieng what do you mean?

**Comment by ultimaweapon** [comment]
@Mou-Ikkai the memory usage depends on the game. It is very unlikely to reach 8 GB. Some simple 2D games may use just a couple hundred megabytes.

**Comment by bayedieng** [comment]
> @bayedieng what do you mean?

You are targeting ```unix``` for your syscall functions which linux and macOS are both apart of. The issue here is that macOS sometimes doesn't support the same flag/variable for the same syscall so it's better to either target linux and macos individually instead of unix.

**Comment by ultimaweapon** [comment]
We can use the same for most POSIX APIs, including this case. It is quite rare to have a separated API between Linux and macOS.

**Comment by bayedieng** [comment]
Yeah, I'm fine with using unix for most syscalls, in this case however, I don't think the sysconf will work because macOS does does not support the flag you are using.

**Comment by ultimaweapon** [comment]
Interesting. I tried to find some information about this but did not find how to get page size on macOS.

**Comment by ultimaweapon** [comment]
I believe it is working according to this issue: https://github.com/rust-lang/miri/issues/216

**Comment by ultimaweapon** [comment]
It works on the iPhone, which means it should work on macOS too: https://developer.apple.com/forums/thread/47532

**Comment by bayedieng** [comment]
I see Apple's online syscall  manpages seem to be outdated. Should be good to merge then.

**Review by bayedieng — APPROVED** [review]

**Pull request merged**